run_model.py

```diff
@@ -18,67 +21,186 @@
 def get_git_revision_hash():
     return subprocess.check_output(['git', 'rev-parse', 'HEAD']).strip().decode('utf8')
 
+def generate_server_configs(server_config, analysis_config):
+    configs_to_analyze = []
+    for analysis_field in analysis_config:
+        for usage_field in server_config["usage"]:
+            if usage_field == analysis_field:
+                model_params = generate_model_params(server_config["usage"][usage_field], analysis_config[analysis_field], usage_field)
+                # Generate a different cluster config for each entry in model_params
+                for model_param in model_params:
+                    analysis_model_dict = copy.deepcopy(server_config)
+                    analysis_model_dict["usage"][usage_field] = model_param['model']
+                    # Write the config to a temp file so it can be make into a ClusterConfig object
+                    with open(tempfile.NamedTemporaryFile().name, 'a+') as f:
+                        f.write(json.dumps(analysis_model_dict))
+                        f.seek(0)
+                        analysis_model_config = cluster_config_from_path(f.name)
+                    configs_to_analyze.append({"model": analysis_model_config,
+                                               "field_to_modify": model_param["field_to_modify"],
+                                               "subfield_to_modify": model_param["subfield_to_modify"],
+                                               "step_value": model_param["step_value"]})
+
+    return configs_to_analyze
+
+
+def generate_model_params(model_params, analysis_config_entry, field_to_modify):
+    step_values = generate_steps(analysis_config_entry)
+    analysis_model_params = []
+    subfield_to_modify = analysis_config_entry["field_name"]
+    for step_value in step_values:
+        this_model_def = copy.deepcopy(model_params)
+        this_model_def[subfield_to_modify] = step_value * model_params[subfield_to_modify]
+        analysis_model_params.append({"model": this_model_def,
+                                      "field_to_modify": field_to_modify,
+                                      "subfield_to_modify": subfield_to_modify,
+                                      "step_value": step_value})
+    return analysis_model_params
+
+def generate_steps(analysis_config_entry):
```

Comment:
should move these functions to a separate file

Reply:
I moved these to the separate entry point and marked as internal: https://github.com/dimagi/commcare_resource_model/pull/25/commits/20842292c290b50ac2104cbc06c2255a11dc0be3

Do you think I should just move them to a separate file instead?